docs/public/specifications/tests/Nameserver-TP/nameserver15.md

```diff
@@ -75,22 +83,24 @@ servers.
     2. If there is no DNS response, then go to next name server IP.
     3. Add the name server IP to the *Sending Version Query* set.
     4. For each query name in *Query Names* do:
-       1. [Send] *TXT Query* with query name to
-          the name server and collect the response.
+       1. [Send] *TXT Query* with query name to the name server and collect the
+          response.
        2. If there is no DNS response or the response has the [RCODE Name]
           ServFail, add name server and query name to the
           *Error On Version Query* set and go to next query name.
        3. If the [DNS Response] does not have any TXT record in the answer
           section with query name as owner name, go to next query name.
-       4. For each TXT record in the answer section of the [DNS Response]
-          with owner name query name, extract and [concatenate] the string(s)
-          from the RDATA of the record.
-       5. If the extracted string is non-empty, add *Name Server IP*, query name
-          and the string to the *TXT Data* set.
-
-7.  If the *TXT Data* set is non-empty, then, for each unique
-    string and query name pair in the set, output *[N15_SOFTWARE_VERSION]*
-    with name server IP list, query name and string.
+       4. For each TXT record in the answer section of the [DNS Response] do:
+          1. If [DNS Class] of the TXT record is not CH, then add name server
+             to the *Wrong Record Class* set.
```

Comment:
Just one suggestion, in order to address #1208 properly:
```suggestion
          1. If [DNS Class] of the TXT record is not CH, then add the name server
             to the *Wrong Record Class* set and go to next TXT record.
```

Reply:
Since that is in a loop "for each TXT record" what would "go to next TXT record" add?

Reply:
Sometimes there is too much magic behind the scene. The idea is to just continue if the TXT record is in, say, IN class. That should work, should it? Unless some code behind filters non-CH responses.